bambi/results.py

```diff
@@ -114,19 +132,96 @@ def plot(self, burn_in=0, names=None, annotate=True, hide_transformed=True,
+        if self.model.family.name=='binomial' and np.max(self.model.y.data) < 1.01:
+            event = next(i for i,x in enumerate(self.model.y.data.flatten()) if x==1)
+            print('NOTE: Modeling the probability that {}==\'{}\''.format(
+                self.model.y.name, str(self.model.data[self.model.y.name][event])))
 
         return ax
 
-    def summary(self, burn_in=0, fixed=True, random=True, names=None,
+    def summary(self, burn_in=0, exclude_ranefs=True, names=None,
                 hide_transformed=True, **kwargs):
         '''
-        Summarizes all parameter estimates. Currently just a wrapper for
-        pm.summary().
+        Summarizes all parameter estimates. Basically a wrapper for
+        pm.df_summary() plus some niceties.
+        Args:
+            burn_in (int): Number of initial samples to exclude before
+                summary statistics are computed.
+            exclude_ranefs (bool): If True (default), do not print
+                summary statistics for individual random effects.
+            names (list): Optional list of variable names to summarize.
+            hide_transformed (bool): If True (default), do not print
+                summary statistics for internally transformed variables.
+        '''
+
+        # if no 'names' specified, filter out unwanted variables
+        if names is None:
+            names = self.untransformed_vars if hide_transformed else self.trace.varnames
+            if exclude_ranefs:
+                names = [x for x in names
+                    if x[2:] not in list(self.model.random_terms.keys())]
+
+        # get the basic DataFrame
+        df = pm.df_summary(self.trace[burn_in:], varnames=names, **kwargs)
+
+        # replace the "__\d" suffixes with an informative factor level name
+        match = [re.match('^(.*)(?:__)(\d+)?$', x) for x in df.index]
+        def replace_with_name(match):
+            term = self.model.terms[match.group(1)[2:]]
+            # handle fixed effects
+            if term in self.model.fixed_terms.values():
+                return term.levels[int(match.group(2))]
+            # handle random effects
+            else:
+                return '{}[{}]'.format(term.name, term.levels[int(match.group(2))])
+        new = [replace_with_name(x) if x is not None else df.index[i]
+            for i,x in enumerate(match)]
+        df.set_index([new], inplace=True)
+
+        # For binomial models with n_trials = 1 (most common use case),
+        # tell user which event is being modeled
+        if self.model.family.name=='binomial' and np.max(self.model.y.data) < 1.01:
+            event = next(i for i,x in enumerate(self.model.y.data.flatten()) if x==1)
+            print('NOTE: Modeling the probability that {}==\'{}\''.format(
+                self.model.y.name, str(self.model.data[self.model.y.name][event])))
+
+        return df
+
+    def get_trace(self, burn_in=0, names=None, hide_transformed=True):
```

Comment:
This is great! But maybe we should add the same `exclude_ranefs` flag as in `summary`, since most of the columns are taken up by individual ranef parameters.